stream-chat-android/src/main/java/com/getstream/sdk/chat/viewmodel/messages/MessageListViewModelBinding.kt

```diff
@@ -26,8 +26,20 @@ public fun MessageListViewModel.bindView(view: MessageListView, lifecycleOwner:
     view.setOnMessageRetryHandler { onEvent(RetryMessage(it)) }
 
     state.observe(lifecycleOwner) { state ->
-        if (state is MessageListViewModel.State.Result) {
-            view.displayNewMessage(state.messageListItem)
+        when (state) {
+            is MessageListViewModel.State.Loading -> {
+                view.hideEmptyStateView()
+                view.showLoadingView()
+            }
+            is MessageListViewModel.State.Result -> {
+                if (state.messageListItem.items.isEmpty()) {
+                    view.showEmptyStateView()
```

Comment:
Do we need two methods for what amounts to boolean logic? If we did `fun showEmptyStateView(show: Boolean = true) { ... }`, this could become any of 
- `view.showEmptyStateView(state.messageListItem.items.isEmpty())`
- ` state.messageListItem.items.isEmpty().let { itemsEmpty -> view.showEmptyStateView(itemsEmpty) }`
- ` state.messageListItem.items.isEmpty().let(view::showEmptyStateView)` 
but could also still be called as `view.showEmptyStateView`.

Just a thought that crossed my mind /shrug.

Reply:
IMO, `showEmptyState(true)` and `showEmtpyState(false)` can be confusing.